orca/src/llm/openai.rs

```diff
@@ -273,25 +274,40 @@ impl EmbeddingTrait for OpenAI {
         Ok(res.into())
     }
 
-    /// TODO: Concurrent
-    async fn generate_embeddings(&self, prompt: Vec<Box<dyn Prompt>>) -> Result<EmbeddingResponse> {
+    async fn generate_embeddings(&self, prompts: Vec<Box<dyn Prompt>>) -> Result<EmbeddingResponse> {
         let mut embeddings = Vec::new();
-        for prompt in prompt {
+        let (sender, receiver) = bounded(prompts.len());
```

Comment:
Instead of using crossbeam, why not use [`tokio::sync::mpsc`](https://docs.rs/tokio/1.33.0/tokio/sync/mpsc/index.html)? This will allow us to await inside of the spawned tokio task, preventing the sender from blocking.

Reply:
Great idea! I still need to learn so much more about all the available crates! 😅